### PR TITLE
Shaders fragment input struct mobile optimizations

### PR DESCRIPTION
**Pull request by manuele-bonanno**
### Purpose of this PR
Following a previous optimization of the Terrain and Forward lighting fragment input struct, this PR applies the same optimization to additional lighting shaders.
This was discussed with Apple and Felipe in the previous PR.

---
### Testing status
local and Yamato

---
### Comments to reviewers

Before:
VS_Pass_Bandwidth_BeforePatch

After:
VS_Pass_Bandwidth_AfterPatch

### CODE REVIEW

**Comment by manuele-bonanno** [comment]
the failed URP tests are also failing on master and don't seem related to this PR

**Review by phi-lira — APPROVED** [review]

File: com.unity.render-pipelines.universal/Shaders/SimpleLitGBufferPass.hlsl [hunk 1]
```diff
@@ -92,25 +97,27 @@ Varyings LitPassVertexSimple(Attributes input)
...
     VertexNormalInputs normalInput = GetVertexNormalInputs(input.normalOS, input.tangentOS);
     half3 viewDirWS = GetWorldSpaceViewDir(vertexInput.positionWS);
-    half3 vertexLight = VertexLighting(vertexInput.positionWS, normalInput.normalWS);
 
     output.uv = TRANSFORM_TEX(input.texcoord, _BaseMap);
     output.posWS.xyz = vertexInput.positionWS;
     output.positionCS = vertexInput.positionCS;
 
 #ifdef _NORMALMAP
+    half3 viewDirWS = GetWorldSpaceViewDir(vertexInput.positionWS);
```
Comment (phi-lira) [inline com.unity.render-pipelines.universal/Shaders/SimpleLitGBufferPass.hlsl:106]: redefinion of viewDirWS

Comment (phi-lira) [reply com.unity.render-pipelines.universal/Shaders/SimpleLitGBufferPass.hlsl:106]: Same on SimpleLit shader.

**Review by phi-lira — CHANGES_REQUESTED** [review]
Fix shader compiler error due to redefinition.

File: com.unity.render-pipelines.universal/Shaders/SimpleLitGBufferPass.hlsl (hunk 1 above)
Comment (manuele-bonanno) [reply com.unity.render-pipelines.universal/Shaders/SimpleLitGBufferPass.hlsl:105]: fixed this and another shader semantics error in the terrain shader

**Comment by manuele-bonanno** [comment]
fixed the shader build related URP tests errors. The current test failures seem consistent with the ones in the nightly master build, so don't seem to be introduced by this PR

**Review by phi-lira — APPROVED** [review]

**Pull request closed without merging**